app/src/main/java/com/codesoom/assignment/application/TaskRepository.java

```diff
@@ -0,0 +1,93 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.models.IdGenerator;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * 할 일을 저장하고, 조회, 수정, 삭제 기능을 제공하는 클래스입니다.
+ */
+@Repository
+public class TaskRepository {
+    private final IdGenerator taskIdGenerator;
+    private Map<Long, Task> taskMap = new ConcurrentHashMap<>();
+
+    public TaskRepository(IdGenerator taskIdGenerator) {
+        this.taskIdGenerator = taskIdGenerator;
+    }
+
+    /**
+     * 할 일 목록 전체를 반환합니다.
+     * @return 할 일 목록
+     */
+    public Collection<Task> getTaskList() {
+        return taskMap.values();
+    }
+
+    /**
+     * 주어진 id를 가진 할 일을 반환합니다.
+     * @param id 할 일 식별자
+     * @return 할 일
+     */
+    public Optional<Task> getTaskById(Long id) {
+        return Optional.ofNullable(taskMap.get(id));
+    }
+
+    /**
+     * 주어진 할 일을 추가합니다.
+     * @param task 할 일
+     * @return 추가된 할 일
+     */
```

Comment:
javadoc을 작성할 때에는 무엇을 리턴하는지만 설명하기만 해도 평타를 칩니다. 꼭 기억해 두세요. 언제나 리턴값을 설명해 주세요.

```suggestion
    /**
     * 주어진 할 일을 추가하고 리턴합니다.
     * @param task 할 일
     * @return 추가된 할 일
     */
```

Reply:
넵 알겠습니다!